pkg/api/steve/supportconfigs/supportconfigs.go

```diff
@@ -0,0 +1,57 @@
+package supportconfigs
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/version"
+)
+
+// supportConfig represents the serializable struct of support related information common to all rancher setups
+type supportConfig struct {
+	// SupportEligible - if this rancher product is eligible for support.
+	SupportEligible bool `json:"support_eligible,omitempty"`
+	// Platform - the platform (arm, x86_64, etc) rancher is running on. As of now, always x86
+	Platform string `json:"platform"`
+	// Product - suse product string identifying rancher - see getRancherProductString
+	Product string `json:"product"`
+}
+
+// supportConfigHandler contains the necessary clients/controllers for retrieving a support config
+type supportConfigHandler struct {
+}
```

Comment:
Looks like we don't need this struct since it doesn't contain anything. Remove if that is the case.

Reply:
I put this here because future uses of the supportconfig (i.e. retrieving logs) definitely will need to have stuff here (i.e. fetching the logs will require some form of k8s client). Because of this, I would prefer to keep it. Let me know if you feel otherwise.

Reply:
It seems unnecessary and I'm not sure we should leave code stubs in hopes that it influences future additions. @cbron  or @cmurphy thoughts?

Reply:
Sorry - I did not update this thread. I have removed this struct - this endpoint no longer generates any values itself (it gets it all from the csp configmap). In light of this, I don't think this thread is relevant any longer. If you disagree, let me know.